src/test/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkTest.java

```diff
@@ -120,7 +119,7 @@ private void testMetric(Job mockJob, long expectedCount, int invocations)
     BigQuerySink sink = getSinkToTest(mockJob);
     MockStageMetrics mockStageMetrics = Mockito.spy(new MockStageMetrics("test"));
     BatchSinkContext context = getContextWithMetrics(mockStageMetrics);
-    sink.onRunFinish(true, context);
+    sink.recordMetric(true, context);
```

Comment:
why is this change needed?

Reply:
Previously, onRunFinish would only execute the filesystem delete call when a temporary bucket was created (e.g. the temporary bucket name was left empty). However, since this change now will execute the delete call after all runs no matter whether a temporary bucket is provided, the unit test will hit that condition and result in a NullPointerException since the Hadoop baseConfiguration doesn't exist.

Since these unit tests only test the recordMetric method anyway, I figured it was cleaner to change it to directly testing that method instead of testing the entire onRunFinish method, since refactoring/mocking the entire Hadoop configuration is nontrivial.

Reply:
I see